### ISSUE TITLE
Outer attributes of method arguments occasionally disappear upon formatting

### ISSUE THREAD

**Issue by alethes** [rust-lang/rustfmt#4338]
I've noticed that `rustfmt` removes certain outer attributes of my methods' arguments while preserving others, which are often almost identical. The bug seems to have something to do with the number of non-whitespace characters used in tokens related to an argument (i.e. attributes, argument identifiers and argument types). E.g., the most reduced example I could come up with is:

```
impl A {
    fn a(#[arg(a = 1)] aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bool) {}
}
```

It produces the following output:
```
impl A {
    fn a(aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bool) {}
}
```

, whereas I would expect:
```
impl A {
    fn a(
        #[arg(a = 1)] aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa: bool,
    ) {
    }
}

```

Removing just a single `a` from the argument's name leads to correct results. Besides using a very long argument name, the issue can be triggered by, among others, a long attribute name, long attribute argument names and values, a large number of attribute arguments, a long argument type identifier, or a combination of those. However, there are combinations with even longer argument declarations that don't lead to this bug. E.g.:
```
impl A {
    fn a(
        #[arg(aaaaaaaaaaaaaaaaaa = "aaaaaaaaaaaaaaaaaaaa", bbbbbbbbbbbbbb = "bbbbbbbbbbbbbbbbbbbbbbb", ccccccccccccccccccccccccccc = "ccccccccccccccccccccccccccccccccc", d = "gweeeeeeeeeeeeeeee")] aadddddddddddddfffffffffffffffffefessssssssssssssfefffffffff: bool,
    ) {
    }
}
```
is formatted correctly. In general, though, increasing the length of an argument's name beyond a certain, varying threshold, appears to be the most reliable way to provoke the bug.

**Meta**

- rustfmt version: rustfmt 1.4.18-nightly (c1e9b7b 2020-06-13) (appears on stable as well)
- From where did you install rustfmt?: rustup
- How do you run rustfmt: rustfmt

**Comment by ayazhafiz** [comment]
This was fixed in bfec1e98e1a7c934c82bf78fc494bfa907558feb, released in [v2.0.0-rc.2](https://github.com/rust-lang/rustfmt/releases/tag/v2.0.0-rc.2)

**Comment by calebcartwright** [comment]
Thanks for the report @alethes and @ayazhafiz for confirming the fix! Closing as a duplicate
